src/common/src/system_param/reader.rs

```diff
@@ -40,87 +42,111 @@ for_all_params!(define_system_params_read_trait);
 ///
 /// See [`SystemParamsRead`] for more details.
 #[derive(Clone, Debug, PartialEq)]
-pub struct SystemParamsReader {
-    prost: PbSystemParams,
+pub struct SystemParamsReader<I = PbSystemParams> {
+    inner: I,
```

Comment:
Why introducing `I` here?

Reply:
Here we want to create a reader with the reference to a `PbSystemParams`:

https://github.com/risingwavelabs/risingwave/blob/3a54d0b1191f310963dcdf09a5de1d98242e0496/src/common/src/system_param/mod.rs#L327